onnxruntime/core/providers/cpu/sequence/sequence_ops.cc

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/cpu/sequence/sequence_ops.h"
+#include "core/framework/tensorprotoutils.h"
+
+using namespace onnxruntime::common;
+
+namespace onnxruntime {
+
+// SequenceLength
+ONNX_CPU_OPERATOR_KERNEL(
+    SequenceLength,
+    11,
+    KernelDefBuilder()
+        .TypeConstraint("S", DataTypeImpl::AllSequenceTensorTypes())
+        .TypeConstraint("I", DataTypeImpl::GetTensorType<int64_t>()),
+    SequenceLength);
+
+Status SequenceLength::Compute(OpKernelContext* context) const {
+  const auto* X = context->Input<VectorTensor>(0);
+  ORT_ENFORCE(X != nullptr, "Got nullptr for sequence input.");
+
+  auto* Y = context->Output(0, {});
+  auto* Y_data = Y->template MutableData<int64_t>();
+  *Y_data = static_cast<int64_t>(X->size());
+
+  return Status::OK();
+}
+
+// SequenceAt
+ONNX_CPU_OPERATOR_KERNEL(
+    SequenceAt,
+    11,
+    KernelDefBuilder()
+        .TypeConstraint("S", DataTypeImpl::AllSequenceTensorTypes())
+        .TypeConstraint("T", DataTypeImpl::AllTensorTypes())
+        .TypeConstraint("I", std::vector<MLDataType>{
+                                 DataTypeImpl::GetTensorType<int32_t>(),
+                                 DataTypeImpl::GetTensorType<int64_t>()}),
+    SequenceAt);
+
+static void GetSeqIdx(const Tensor& idx_tensor, int64_t& seq_idx) {
+  auto idx_tensor_dtype = utils::GetTensorProtoType(idx_tensor);
```

Comment:
Would returning seq_idx be a little more natural than passing it as a non-const argument?

Reply:
I did it this way so that I could throw an exception for an unsupported data type.

Reply:
sure.